File: frontend/src/types/prediction.ts
export interface PredictionRequest {
  carpet_area_clean: number;
  floor_clean: number;
  bathroom_clean: number;
  balcony_clean: number;
  parking_clean: number;

  location_clean: string;
  society_clean: string;
  Status: string;
  Transaction: string;
  Furnishing: string;
  facing: string;
  overlooking: string;
  Ownership: string;
}

export interface PredictionResponse {
  predicted_price: number;
}

File: backend/app/schemas/prediction.py
from pydantic import BaseModel


class PredictionRequest(BaseModel):
    carpet_area_clean: float
    floor_clean: float
    bathroom_clean: float
    balcony_clean: float
    parking_clean: float

    location_clean: str
    society_clean: str
    Status: str
    Transaction: str
    Furnishing: str
    facing: str
    overlooking: str
    Ownership: str


class PredictionResponse(BaseModel):
    predicted_price: float